sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/objects.scala

```diff
@@ -688,3 +688,38 @@ case class AssertNotNull(child: Expression, walkedTypePath: Seq[String])
     ev.copy(code = code, isNull = "false", value = childGen.value)
   }
 }
+
+/**
+ * Returns the value of field at index `index` from the external row `child`.
```

Comment:
Would be nice to add a more intuitive description:

> This class can be viewed as [[GetStructField]] for [[Row]]s instead of [[InternalRow]]s.

It took me some time to realize this...